ts/sagas/wallet.ts

```diff
@@ -2,49 +2,97 @@
  * A saga that manages the Wallet.
  */
 
-import { call, Effect, put, takeLatest } from "redux-saga/effects";
+import { call, Effect, fork, put, select, take } from "redux-saga/effects";
 
+import { Option } from "fp-ts/lib/Option";
+import { addCard, deleteCard } from "../api/pagopa";
 import { WalletAPI } from "../api/wallet/wallet-api";
 import {
+  ADD_CARD_REQUEST,
+  CARD_ERROR,
+  DELETE_CARD_REQUEST,
   FETCH_CARDS_REQUEST,
   FETCH_TRANSACTIONS_REQUEST
 } from "../store/actions/constants";
 import { cardsFetched } from "../store/actions/wallet/cards";
 import { transactionsFetched } from "../store/actions/wallet/transactions";
-import { CreditCard } from "../types/CreditCard";
+import { getNewCardData } from "../store/reducers/wallet/cards";
+import { CreditCard, CreditCardId } from "../types/CreditCard";
 import { WalletTransaction } from "../types/wallet";
 
-function* fetchTransactions(
-  loadTransactions: () => Promise<ReadonlyArray<WalletTransaction>>
-): Iterator<Effect> {
+function* fetchTransactions(_: string): Iterator<Effect> {
   const transactions: ReadonlyArray<WalletTransaction> = yield call(
-    loadTransactions
+    WalletAPI.getTransactions
   );
   yield put(transactionsFetched(transactions));
 }
 
-function* fetchCreditCards(
-  loadCards: () => Promise<ReadonlyArray<CreditCard>>
-): Iterator<Effect> {
-  const cards: ReadonlyArray<CreditCard> = yield call(loadCards);
+function* fetchCreditCards(_: string): Iterator<Effect> {
+  const cards: ReadonlyArray<CreditCard> = yield call(WalletAPI.getCreditCards);
   yield put(cardsFetched(cards));
 }
 
+function* addCardSaga(token: string): Iterator<Effect> {
+  const card: Option<CreditCard> = yield select(getNewCardData);
```

Comment:
I don't understand this step, can you pls add some more comments? I'd like to see [this kind of comments](https://github.com/teamdigitale/italia-app/blob/master/ts/sagas/messages.ts#L108)